src/providers/ad/ad_gpo.c

```diff
@@ -564,6 +564,62 @@ ad_gpo_dom_sid_equal(const struct dom_sid *sid1, const struct dom_sid *sid2)
     return true;
 }
 
+/*
+ * This function retrieves the SID of the group with given gid.
+ */
+static char *
+ad_gpo_get_primary_group_sid(TALLOC_CTX *mem_ctx,
+                             gid_t gid,
+                             struct sss_domain_info *domain,
+                             struct sss_idmap_ctx *idmap_ctx)
+{
+    char *sid = NULL;
+    char *result;
+    const char *attrs[] = {
+        SYSDB_SID_STR,
+        NULL
+    };
+    struct ldb_message *msg;
+    int ret;
+
+    if (gid == 0) {
+        return NULL;
+    }
+
+    ret = sss_idmap_unix_to_sid(idmap_ctx, gid, &sid);
+    if (ret == EOK) {
+        result = talloc_strdup(mem_ctx, sid);
+        sss_idmap_free_sid(idmap_ctx, sid);
+        if (result == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE, "Out of memory while getting SID of the group\n");
+        }
+        return result;
+    }
+
+    if (ret == IDMAP_EXTERNAL) {
+        /* no ID mapping in this domain, search for the group object and get sid there */
+        ret = sysdb_search_group_by_gid(mem_ctx, domain, gid, attrs, &msg);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE, "Search for group '%"SPRIgid"' failded with error '%d'\n", gid, ret);
+            return NULL;
+        }
+
+        sid = discard_const(ldb_msg_find_attr_as_string(msg, SYSDB_SID_STR, NULL));
```

Comment:
Hi,

I would prefer to avoid `discard_const()` even if this would mean to declare another variable like e.g. `const char *tmp;`.

bye,
Sumit

Reply:
done